packages/oruga/src/composables/useInputHandler.ts

```diff
@@ -240,26 +280,29 @@ export function useInputHandler<T extends ValidatableFormElement>(
         // (We're also expecting the use of radio buttons with our default validation message handling
         // to be fairly uncommon because the overall visual experience is clunky with such a configuration.)
         const onAttributeChange = (): void => {
+            updateCustomValidationMessage();
```

Comment:
Would it be nicer to call this within `checkHtml5Validity`? Then it will also be called when the component triggers the call, like when the input value changes.

Reply:
We probably could call it there. My initial thought was that `checkHtml5Validity` wouldn't be called often enough for all cases. If we're just overriding an existing constraint validation message, it's fine to wait until right before we'd display the message (especially since components in a known invalid state should be calling `checkHtml5Validity` on every change). For the case where we're adding a custom constraint validation message to an otherwise valid input, though, we won't call `checkHtml5Validity` until the input receives a `blur` or `invalid` event, which may be too late if the user submits the form using the Enter key.

We might be able to rely on the call to `watchEffect(updateCustomValidationMessage)`, but as noted above, we'd have to ensure that any new reactive effects that show up with a new call to `updateCustomValidationMessage` are observed within the `watchEffect` context. The only way I've found to do that is to trigger the watcher to run its effect again.

Reply:
I found an issue that required me to call `checkHtml5Validity` from inside the `watchEffect`, so I had to flip the order of calls around to avoid infinite recursion.